contracts/paymaster/src/paymaster.rs

```diff
@@ -0,0 +1,49 @@
+#![no_std]
+
+use forward_call::PaymentsVec;
+
+multiversx_sc::imports!();
+
+pub mod forward_call;
+const FEE_PAYMENT: usize = 0;
+
+/// An empty contract. To be used as a template when starting a new contract from scratch.
+#[multiversx_sc::contract]
+pub trait PaymasterContract: forward_call::ForwardCall {
+    #[init]
+    fn init(&self) {}
+
+    #[endpoint(forwardExecution)]
+    #[payable("*")]
+    fn forward_execution(
+        &self,
+        relayer_addr: ManagedAddress,
+        dest: ManagedAddress,
+        endpoint_name: ManagedBuffer,
+        endpoint_args: MultiValueEncoded<ManagedBuffer>,
+    ) {
+        let payments = self.call_value().all_esdt_transfers();
+
+        self.pay_fee_to_relayer(relayer_addr, payments.clone_value());
```

Comment:
payments.first.clone_calue ? no need to give all the list in this function.

Reply:
Done